lib/routemap.c

```diff
@@ -1461,14 +1461,14 @@ int route_map_delete_set(struct route_map_index *index, const char *set_name,
    (note, this includes the description for the "NEXT"
    and "GOTO" frobs now
 
-	      Match   |   No Match
-		      |
-    permit    action  |     cont
-		      |
-    ------------------+---------------
-		      |
-    deny      deny    |     cont
-		      |
+	   |   Match   |   No Match   |  ok
```

Comment:
This is a fundamental change in how we handle routemaps, Please put this into it's own commit with your detailed comments that were in the PR itself as too your reasoning and what you were thinking.

I'm personally a bit leary that we are using RMAP_OKAY to mean different things based on if this is a match or set part of a route map.  Seems like a recipe for an easy screw up down the line.  It almost seems like we need a `enum route_map_match_result` and a `enum route_map_set_result` and go through and split everything up in the code.

Thoughts from other reviewers?